src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -223,6 +261,74 @@ public class RestartComputerCommand : PSCmdlet, IDisposable
         [Alias("f")]
         public SwitchParameter Force { get; set; }
 
+        /// <summary>
+        /// Provide a comment for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        public string Comment { get; set; } = "";
+
+        /// <summary>
+        /// The major reason for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateSet(
+            "Application",
+            "Hardware",
+            "OperatingSystem",
+            "Other",
+            "Power",
+            "Software",
+            "System")] 
+        public string MajorReason { get; set; }
+
+        /// <summary>
+        /// The minor reason for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateSet(
+            "BlueScreen",
+            "Disk",
+            "Environment",
+            "Driver",
+            "HotFix",
+            "HotFixUninstall",
+            "Unresponsive",
+            "Installation",
+            "Maintenance",
+            "MMC",
+            "NetworkConnectivity",
+            "NetworkCard",
+            "Other",
+            "OtherDriver",
+            "PowerSupply",
+            "Processor",
+            "Reconfigure",
+            "SecurityIssue",
+            "SecurityPatch",
+            "SecurityPatchUninstallation",
+            "ServicePack",
+            "ServicePackUninstallation",
+            "TerminalServices",
+            "Unstable",
+            "Upgrade",
+            "WMI")]
+        public string MinorReason { get; set; }
+
+        /// <summary>
+        /// Specify a delay before the reboot occurs.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateRange(0, UInt32.MaxValue)]
+        public UInt32 RestartDelay
+        {
+            get { return _restartDelay; }
+            set
+            {
+                _restartDelay = value;
+            }
+        }
+        private UInt32 _restartDelay = 0;
```

Comment:
Please remove and use auto-property `public UInt32 RestartDelay { get; set; }`.
No need to init - default is 0.